pkg/network/server.go

```diff
@@ -86,6 +86,13 @@ type (
 		Shutdown()
 	}
 
+	// BlockQueuer is an abstraction of the block queue that is completely
+	// managed by server (including queue runs and discards). The interface
+	// exposes a single method sufficient for the subsequent block enqueueing.
+	BlockQueuer interface {
```

Comment:
I don't think you need this interface.

Reply:
I've added to resrict the rights of the users of server's block queue. Otherwise our network server exposes its block queue with `Run` and `Discard` methods, so any other user of the server's blockqueue potentially may discard it, which should never happen.

Removed.